engine/transaction/record/testabilities/mock_repository.go

```diff
@@ -1,31 +1,30 @@
-package record_test
+package testabilities
 
 import (
 	"context"
 	"iter"
 	"maps"
 	"slices"
 
-	trx "github.com/bitcoin-sv/go-sdk/transaction"
 	"github.com/bitcoin-sv/spv-wallet/engine/database"
 	"github.com/bitcoin-sv/spv-wallet/models/bsv"
 )
 
-type mockRepository struct {
+type MockRepository struct {
 	transactions map[string]database.Transaction
 	outputs      map[string]database.Output
 	data         map[string]database.Data
 }
 
-func newMockRepository() *mockRepository {
-	return &mockRepository{
+func NewMockRepository() *MockRepository {
+	return &MockRepository{
 		transactions: make(map[string]database.Transaction),
 		outputs:      make(map[string]database.Output),
 		data:         make(map[string]database.Data),
 	}
 }
 
-func (m *mockRepository) SaveTX(_ context.Context, txTable *database.Transaction, outputs []*database.Output, data []*database.Data) error {
+func (m *MockRepository) SaveTX(_ context.Context, txTable *database.Transaction, outputs []*database.Output, data []*database.Data) error {
```

Comment:
Maybe some testabilities like:
WillFailOnSaveTX as a valuable test case? + such example in `record_outline_test.go`?

Reply:
Good idea 
I'm adding this `WillFailOnSaveTX ` and a test case for that 🚀

Reply:
Done 😄